django/core/mail/message.py

```diff
@@ -71,56 +72,57 @@ def forbid_multi_line_headers(name, val, encoding):
     return name, val
 
 
-def split_addr(addr, encoding):
-    """
-    Split the address into local part and domain and encode them.
-
-    When non-ascii characters are present in the local part, it must be
-    MIME-word encoded. The domain name must be idna-encoded if it contains
-    non-ascii characters.
-    """
-    if '@' in addr:
-        localpart, domain = addr.split('@', 1)
-        # Try to get the simplest encoding - ascii if possible so that
-        # to@example.com doesn't become =?utf-8?q?to?=@example.com. This
-        # makes unit testing a bit easier and more readable.
-        try:
-            localpart.encode('ascii')
-        except UnicodeEncodeError:
-            localpart = Header(localpart, encoding).encode()
-        domain = domain.encode('idna').decode('ascii')
-    else:
-        localpart = Header(addr, encoding).encode()
-        domain = ''
-    return (localpart, domain)
-
-
 def sanitize_address(addr, encoding):
     """
     Format a pair of (name, address) or an email address string.
     """
+    address = None
+
     if not isinstance(addr, tuple):
-        addr = parseaddr(addr)
-    nm, addr = addr
-    localpart, domain = None, None
+        addr = force_str(addr)
+        try:
+            # This will try either to parse 'Name <add@ress>' or 'add@ress'
+            # get_mailbox raises an IndexError when given "a@"
+            # and a HeaderParseError when given "@"
+            token, rest = parser.get_mailbox(addr)
+            if not isinstance(token, Mailbox):
+                raise ValueError
+        except (HeaderParseError, ValueError, IndexError):
```

Comment:
I'm wondering whether "except Exception" might be better suited here. Whatever get_mailbox() finds is enough to indicate the address should be refused, and we probably don't want an exception to bubble here I'd guess.